src/main/scala/exercises/05_Patterns.scala

```diff
@@ -55,16 +61,31 @@ object PatternMatching {
    *        
    *    - anything else, the function result is "Some Scala class"
    */
-  def matchOnInputType(in: Any): String = {
-    error("fix me")
+  def matchOnInputType(in: Any): String = in match {
+    case s: String => "A string with length " + s.length.toString
+    case _: Integer => "A positive integer"
+    case p: Person => "A person with name: " + p.name
+    case seq1@Seq(hd1, hd2, tail @ _ *) => if (seq1.length > 10) {
```

Comment:
Very sophisticated use of binding matched patterns to variables to handle both the sequence cases simultaneously. Very nice!